modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/exec/row/SqlRowHandlerTest.java

```diff
@@ -0,0 +1,212 @@
+
+        RowWrapper concatenated = handler.concat(params.left, params.right);
+
+        int leftLen = params.leftData.length;
+        int rightLen = params.rightData.length;
+        int totalElementsCount = leftLen + rightLen;
+
+        assertThat(handler.columnCount(concatenated), equalTo(totalElementsCount));
+
+        // Build combined schema.
+        Builder builder = RowSchema.builder();
+        params.leftSchema.fields().forEach(builder::addField);
+        params.rightSchema.fields().forEach(builder::addField);
+
+        RowSchema concatenatedSchema = builder.build();
+
+        // Serialize.
+        ByteBuffer buf = handler.toByteBuffer(concatenated);
+
+        // Wrap into row.
+        RowWrapper result = handler.factory(concatenatedSchema).wrap(new BinaryTuple(totalElementsCount, buf));
+
+        for (int i = 0; i < leftLen; i++) {
+            assertThat(handler.get(i, result), equalTo(TypeUtils.toInternal(params.leftData[i])));
+        }
+
+        for (int i = 0; i < rightLen; i++) {
+            assertThat(handler.get(leftLen + i, result), equalTo(TypeUtils.toInternal(params.rightData[i])));
+        }
+    }
+
+    private static Stream<Arguments> concatTestArguments() {
+        return Stream.of(
+                Arguments.of(Named.of("array", false), Named.of("array", false)),
+                Arguments.of(Named.of("array", false), Named.of("tuple", true)),
+                Arguments.of(Named.of("tuple", true), Named.of("array", false)),
+                Arguments.of(Named.of("tuple", true), Named.of("tuple", true))
+        );
+    }
+
+    private RowSchema rowSchema(List<ColumnType> columnTypes, Object[] values) {
+        Builder schemaBuilder = RowSchema.builder();
+
+        for (int i = 0; i < values.length; i++) {
+            ColumnType type = columnTypes.get(i);
+
+            if (type == ColumnType.NULL) {
+                schemaBuilder.addField(new NullTypeSpec());
+
+                continue;
+            }
+
+            NativeType nativeType = values[i] == null
+                    ? columnType2NativeType(type, 9, 3, 20)
+                    : NativeTypes.fromObject(values[i]);
+
+            schemaBuilder.addField(nativeType, values[i] == null || rnd.nextBoolean());
+        }
+
+        return schemaBuilder.build();
+    }
+
+    private Object[] values(List<ColumnType> columnTypes) {
+        Object[] values = new Object[columnTypes.size()];
+        int baseValue = rnd.nextInt();
+
+        for (int i = 0; i < values.length; i++) {
+            ColumnType type = columnTypes.get(i);
+
+            values[i] = type == ColumnType.NULL ? null : generateValueByType(baseValue, type);
+        }
+
+        return values;
+    }
+
+    private List<ColumnType> columnTypes() {
+        List<ColumnType> columnTypes = new ArrayList<>(EnumSet.complementOf(EnumSet.of(ColumnType.PERIOD, ColumnType.DURATION)));
```

Comment:
it would be better to add a link to an issue with INTERVAL type support: https://issues.apache.org/jira/browse/IGNITE-15200

Reply:
Done, thanks.